lib/netext/transport.go

```diff
@@ -0,0 +1,102 @@
+package netext
+
+import (
+	"net"
+	"net/http"
+	"strconv"
+
+	"github.com/loadimpact/k6/lib"
+	"github.com/loadimpact/k6/stats"
+	"github.com/pkg/errors"
+)
+
+type Transport struct {
+	http.RoundTripper
+	options   lib.Options
+	tags      map[string]string
+	trail     *Trail
+	tlsInfo   TLSInfo
+	samplesCh chan<- stats.SampleContainer
+}
+
+func NewTransport(transport http.RoundTripper, samplesCh chan<- stats.SampleContainer, options lib.Options, tags map[string]string) *Transport {
```

Comment:
If we can't have the whole state because of the import cycles, can we at least make the `Options` variable a pointer? 
`unsafe.Sizeof(lib.Options{})` gives me 440, ideally k6 won't have to copy almost half a kilobyte of the same static options for every HTTP request it makes.